opamp-client/src/main/java/io/opentelemetry/opamp/client/internal/OpampClient.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opamp.client.internal;
+
+import io.opentelemetry.opamp.client.internal.response.MessageData;
+import opamp.proto.Opamp;
+
+public interface OpampClient {
+
+  /**
+   * Starts the client and begin attempts to connect to the Server. Once connection is established
+   * the client will attempt to maintain it by reconnecting if the connection is lost. All failed
+   * connection attempts will be reported via {@link Callback#onConnectFailed(OpampClient,
+   * Throwable)} callback.
+   *
+   * <p>This method does not wait until the connection to the Server is established and will likely
+   * return before the connection attempts are even made.
+   *
+   * <p>This method may be called only once.
+   *
+   * @param callback The Callback to which the Client will notify about any Server requests and
+   *     responses.
+   */
+  void start(Callback callback);
+
+  /**
+   * Stops the client. May be called only after {@link #start(OpampClient.Callback)}. May be called
+   * only once. After this call returns successfully it is guaranteed that no callbacks will be
+   * called. Once stopped, the client cannot be started again.
+   */
+  void stop();
+
+  /**
+   * Sets the current remote config status which will be sent in the next agent to server request.
+   *
+   * @param remoteConfigStatus The new remote config status.
+   */
+  void setRemoteConfigStatus(Opamp.RemoteConfigStatus remoteConfigStatus);
+
+  /**
+   * Sets the current effective config which will be sent in the next agent to server request.
+   *
+   * @param effectiveConfig The new effective config.
+   */
+  void setEffectiveConfig(Opamp.EffectiveConfig effectiveConfig);
+
+  interface Callback {
```

Comment:
Nit: Callback or Callbacks? We use [Callbacks in go](https://pkg.go.dev/github.com/open-telemetry/opamp-go@v0.16.0/client/types#Callbacks).

Reply:
Good point, I'll make the changes 👍